### DISCUSSION TITLE
Filter foreign tables through join

### DISCUSSION THREAD

**Discussion by krish-adi** [supabase/supabase#12900] [Questions] [answered]
I want to be able to filter through foreign tables through a join. I am able to query, but not able to filter. 

Consider these tables:

**table_a**
| id | a_name |
|-|-|
| 1  | a | 

**table_b**
| id | b_name |
|-|-|
| 1  | b | 

**table_c**
| id | c_name | b_id |
|-|-|-|
| 1  | c | 1 |

where foreign key b_id references the **Table B**

**table_ab**
| id | a_id | b_id |
|-|-|-|
| 1  | 1 | 1 |

I want to fetch rows from **Table C** based on **a_id**. 

I tried this, but doesn't seem to give the results I am looking for:

```
const { data, error } = await supabase
                    .from("table_c")
                    .select(
                        "id, c_name, table_b(table_ab(a_id))"
                    )
                    .eq(
                        "table_b.table_ab.a_id)",
                        1
                    );
```

**Comment by GaryAustin1** [comment]
I'm going to move this to discussion for right now as it is either you doing something wrong, or trying to do something not supported ... most likely.  It would help to get someone here in discussion and or in the Supabase Discord to look and recommend things.

For instance I don't believe you can go more than one foreign table deep in your filter.

**Comment by steve-chavez** [comment, marked as answer]
Try with:

```js
const { data, error } = await supabase
                    .from("table_c")
                    .select(
                        "id, c_name, table_b!inner(table_ab!inner(a_id))"
                    )
                    .eq(
                        "table_b.table_ab.a_id",
                        1
                    );
```

Going more than one level deep when filtering should work.

**Reply by krish-adi** [reply]
This works! thanks for the quick reply :D

**Reply by rotimi-best** [reply]
Thanks, this also helped me. I didn't know of the `inner`

**Reply by laubonghaudoi** [reply]
@steve-chavez  I don't think this `inner` keyword is mentioned anywhere in the doc. Is this an api from PostgREST? I can't find this in their doc either.

**Reply by GaryAustin1** [reply]
@laubonghaudoi 

The inner keyword is covered in the PostgREST docs and Supabase select doc also shows it.

https://postgrest.org/en/stable/api.html#embedding-with-top-level-filtering has !inner in it.

![image](https://user-images.githubusercontent.com/54564956/234148198-469bc587-8d2d-4b1d-8e91-1442d6833e07.png)
https://postgrest.org/en/stable/api.html#embedding-with-top-level-filtering has !inner in it.

**Reply by laubonghaudoi** [reply]
Thank you! I tried searching in the supabase-js doc site but it returned no results. Maybe it should be made more visible. Also putting more links to the PostgRESQL doc.

**Reply by thiagomendonca-eu** [reply]
![image](https://github.com/supabase/supabase/assets/66015748/66860b1e-247c-4d3c-85d0-bbc22c4935b4)

It is possible to find an example, which cites the need for "!inner" to filter the parent table rows, in the subbase documentation at: https://subase.com/docs/reference/javascript/select
